sdk/core/core-rest-pipeline/src/nodeHttpClient.ts

```diff
@@ -244,28 +250,35 @@ class NodeHttpClient implements HttpClient {
   }
 
   private getOrCreateAgent(request: PipelineRequest, isInsecure: boolean): http.Agent {
-    if (!request.disableKeepAlive) {
-      if (isInsecure) {
-        if (!this.httpKeepAliveAgent) {
-          this.httpKeepAliveAgent = new http.Agent({
-            keepAlive: true,
-          });
-        }
+    if (!request.tlsSettings && request.disableKeepAlive) {
+      return isInsecure ? http.globalAgent : https.globalAgent;
+    }
+    const settingsIndex = request.tlsSettings ?? {};
```

Comment:
suggestion: should we have a module constant that is something like

```
const DEFAULT_TLS_SETTINGS: TlsSettings = {};
```

So we can detect when one isn't set here, e.g. maybe restructure the above a bit:

```suggestion
    const settingsIndex = request.tlsSettings ?? DEFAULT_TLS_SETTINGS;
    if ((!settingsIndex || settingsIndex === DEFAULT_TLSSETTINGS) && request.disableKeepAlive) {
        return isInsecure ? http.globalAgent : https.globalAgent;
    }
```

Also this will help us a bit by keeping the object ref the same when one isn't specified